From dcf103406ddfacdaa07f5364808b3fb9fdb36590 Mon Sep 17 00:00:00 2001
From: Alex Deucher <alexander.deucher@amd.com>
Date: Wed, 3 Jul 2013 15:14:25 -0400
Subject: [PATCH 713/904] drm/radeon/sumo: implement support for
 disable_gfx_power_gating_in_uvd flag

commit 338a95a95508537e23c82d59a2d87be6fde4b6ff upstream

Some asic revisions need to disable PG when UVD is active.

Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/radeon/sumo_dpm.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/radeon/sumo_dpm.c b/drivers/gpu/drm/radeon/sumo_dpm.c
index b13448f..dc59906 100644
--- a/drivers/gpu/drm/radeon/sumo_dpm.c
+++ b/drivers/gpu/drm/radeon/sumo_dpm.c
@@ -824,7 +824,9 @@ static void sumo_setup_uvd_clocks(struct radeon_device *rdev,
 	radeon_set_uvd_clocks(rdev, new_rps->vclk, new_rps->dclk);
 
 	if (pi->enable_gfx_power_gating) {
-		sumo_gfx_powergating_enable(rdev, true);
+		if (!pi->disable_gfx_power_gating_in_uvd ||
+		    !r600_is_uvd_state(new_rps->class, new_rps->class2))
+			sumo_gfx_powergating_enable(rdev, true);
 	}
 }
 
-- 
1.8.4.93.g57e4c17

